src/libsystemd-network/dhcp-option.c

```diff
@@ -121,6 +121,38 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
         return 0;
 }
 
+int dhcp_option_remove_option(uint8_t options[], size_t length, uint8_t option_code) {
+        size_t offset = 0;
+        uint8_t code;
+
+        while (offset < length) {
+                code = options[offset ++];
+
+                switch (code) {
+                case SD_DHCP_OPTION_PAD:
+                        continue;
+
+                case SD_DHCP_OPTION_END:
+                        return -ENOMSG;
```

Comment:
`-ENOENT` or `0`?

Reply:
Thanks, that makes sense. The body of the function dhcp_option_remove_option is in 
 general a copy-paste of function parse_options. May be to apply these fixes to it also?

Reply:
For `parse_options()`, returning 0 on `SD_DHCP_OPTION_END` is ok. I do not think we need to fix that.